addons/mrp/models/mrp_workorder.py

```diff
@@ -587,53 +587,56 @@ def add_unavailability(row, workcenter_id=None):
 
     def button_start(self):
-        self.ensure_one()
-        if any(not time.date_end for time in self.time_ids.filtered(lambda t: t.user_id.id == self.env.user.id)):
-            return True
-        # As button_start is automatically called in the new view
-        if self.state in ('done', 'cancel'):
-            return True
-
-        if self.product_tracking == 'serial' and self.qty_producing == 0:
-            self.qty_producing = 1.0
-        elif self.qty_producing == 0:
-            self.qty_producing = self.qty_remaining
-
-        if self._should_start_timer():
-            self.env['mrp.workcenter.productivity'].create(
-                self._prepare_timeline_vals(self.duration, datetime.now())
-            )
-
-        if self.production_id.state != 'progress':
-            self.production_id.write({
-                'date_start': datetime.now(),
-            })
-        if self.state == 'progress':
-            return True
-        date_start = fields.Datetime.now()
-        vals = {
-            'state': 'progress',
-            'date_start': date_start,
-        }
-        if not self.leave_id:
-            leave = self.env['resource.calendar.leaves'].create({
-                'name': self.display_name,
-                'calendar_id': self.workcenter_id.resource_calendar_id.id,
-                'date_from': date_start,
-                'date_to': date_start + relativedelta(minutes=self.duration_expected),
-                'resource_id': self.workcenter_id.resource_id.id,
-                'time_type': 'other'
-            })
-            vals['date_finished'] = leave.date_to
-            vals['leave_id'] = leave.id
-            return self.write(vals)
-        else:
-            if not self.date_start or self.date_start > date_start:
-                vals['date_start'] = date_start
-                vals['date_finished'] = self._calculate_date_finished(date_start)
-            if self.date_finished and self.date_finished < date_start:
-                vals['date_finished'] = date_start
-            return self.with_context(bypass_duration_calculation=True).write(vals)
+        for wo in self:
+            if wo.working_state == 'blocked':
+                raise UserError(_('Some workorders require another workorder to be completed first'))
+            if any(not time.date_end for time in wo.time_ids.filtered(lambda t: t.user_id.id == self.env.user.id)):
+                return True
+            # As button_start is automatically called in the new view
+            if wo.state in ('done', 'cancel'):
+                return True
+
+            if wo.product_tracking == 'serial' and wo.qty_producing == 0:
+                wo.qty_producing = 1.0
+            elif wo.qty_producing == 0:
+                wo.qty_producing = wo.qty_remaining
+
+            if wo._should_start_timer():
+                self.env['mrp.workcenter.productivity'].create(
+                    wo._prepare_timeline_vals(wo.duration, datetime.now())
```

Comment:
Some code cleanup, we should use `fields.Datetime.now()` instead of datetime.now()
```suggestion
                    wo._prepare_timeline_vals(wo.duration, fields.Datetime.now())
```